ctms/schemas/contact.py

```diff
@@ -27,6 +28,94 @@
     validate_waitlist_newsletters,
 )
 
+if TYPE_CHECKING:
+    from models import Email
+
+
+def _subscriptions_by_product(subscriptions):
+    by_product = defaultdict(list)
+
+    for subscription in subscriptions:
+        for item in subscription.subscription_items:
+            price = item.price
+            product_data = {
+                "payment_service": "stripe",
+                ###
+                # These come from the Payment Method, not imported from Stripe.
+                "payment_type": None,
+                "card_brand": None,
+                "card_last4": None,
+                "billing_country": None,
+                ###
+                "status": subscription.status,
+                "created": subscription.stripe_created,
+                "start": subscription.start_date,
+                "current_period_start": subscription.current_period_start,
+                "current_period_end": subscription.current_period_end,
+                "canceled_at": subscription.canceled_at,
+                "cancel_at_period_end": subscription.cancel_at_period_end,
+                "ended_at": subscription.ended_at,
+                "product_id": price.stripe_product_id,
+                "product_name": None,  # Products are not imported
+                "price_id": price.stripe_id,
+                "currency": price.currency,
+                "amount": price.unit_amount,
+                "interval_count": price.recurring_interval_count,
+                "interval": price.recurring_interval,
+            }
+            by_product[price.stripe_product_id].append(product_data)
+    return by_product
+
+
+def _product_metadata(subscriptions_by_product):
```

Comment:
Adding a comment may help here, at least explaining that we pick the latest product from the subscription dict values

Reply:
@leplatrem ended up going in a pretty different direction for these functions, so retagging you for review.